tests/endpoints/test_off_topic_channel_names.py

```diff
@@ -0,0 +1,245 @@
+        """Return error message when passing not integer as `random_items`."""
+        url = app.url_path_for("get_off_topic_channel_names")
+        response = client.get(
+            f"{url}?random_items=totally-a-valid-integer", headers=AUTH_HEADER
+        )
+
+        assert response.status_code == 400
+        assert response.json(), {
+            "error": [
+                {
+                    "loc": ["query", "random_items"],
+                    "msg": "value is not a valid integer",
+                    "type": "type_error.integer",
+                }
+            ]
+        }
+
+    def test_returns_400_for_negative_random_items_param(self):
+        """Return error message when passing negative int as `random_items`."""
+        url = app.url_path_for("get_off_topic_channel_names")
+        response = client.get(f"{url}?random_items=-5", headers=AUTH_HEADER)
+
+        assert response.status_code == 404
+        assert response.json(), {
+            "error": ["'random_items' must be a positive integer."]
+        }
+
+
+class TestListView:
+    @classmethod
+    def setup_method(cls):
+        cls.test_name = OffTopicChannelName(name="lemons-lemonade-stand", used=True)
+        cls.test_name_2 = OffTopicChannelName(name="bbq-with-bisk", used=True)
+
+        def override_dependency():
+            mocked_session = Mock()
+            mocked_session.query().all().order_by.return_value = [
+                cls.test_name,
+                cls.test_name_2,
+            ]
+            mocked_session.query().all.return_value = [cls.test_name, cls.test_name_2]
+
+            return mocked_session
+
+        app.dependency_overrides[create_database_session] = override_dependency
+
+    @classmethod
+    def teardown_method(cls):
+        app.dependency_overrides = {}
+
+    def test_returns_name_in_list(self):
+        """Return all off-topic channel names."""
+        url = app.url_path_for("get_off_topic_channel_names")
+        response = client.get(url, headers=AUTH_HEADER)
+
+        assert response.status_code == 200
+        assert response.json(), [self.test_name.name, self.test_name_2.name]
+
+    def test_returns_single_item_with_random_items_param_set_to_1(self):
+        """Return not-used name instead used."""
+        url = app.url_path_for("get_off_topic_channel_names")
+        response = client.get(f"{url}?random_items=1", headers=AUTH_HEADER)
+
+        assert response.status_code == 200
+        assert len(response.json()) == 1
+        assert response.json(), [self.test_name.name]
+
+    def test_running_out_of_names_with_random_parameter(self):
+        """Reset names `used` parameter to `False` when running out of names."""
+        url = app.url_path_for("get_off_topic_channel_names")
+        response = client.get(f"{url}?random_items=2", headers=AUTH_HEADER)
+
+        assert response.status_code == 200
+        assert response.json(), [self.test_name.name, self.test_name_2.name]
```

Comment:
I believe, you should use `assertCountEqual` from unittest to properly test it out.